### PR TITLE
add tests

### PR DESCRIPTION
**Pull request by KarenVentura**
add tests for tags and  for helpers.

@anymoto @FerPerales

### CODE REVIEW

File: app/models/internal_post.rb [hunk 1]
```diff
@@ -8,7 +8,7 @@ class InternalPost < Post
     self.body_html = parser.render(body_markdown)
   end
 
-  after_create :notify_twitter
+  # after_create :notify_twitter
```
Comment (FerPerales) [inline app/models/internal_post.rb]: This shouldn't be commented

File: spec/helpers/post_location_helper_spec.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    end
+  end
+
+  describe "#show_tags" do
+    it "return questions tags" do
+      post = FactoryGirl.create(:external_post)
+      expect(helper.show_tags(post)).to match("question")
+    end
+
+    it "return ruby tag" do
```
Comment (FerPerales) [inline spec/helpers/post_location_helper_spec.rb]: _returns_

File: spec/helpers/post_location_helper_spec.rb [hunk 3]
```diff
@@ -0,0 +1,26 @@
+require "rails_helper"
+describe PostLocationHelper, :type => :helper do
+  describe "#link_to_post" do
+    it "returns gamedev url" do
+      post = FactoryGirl.create(:external_post)
+      expect(helper.link_to_post(post)).to match("http://gamedev.stackexchange.com/")
+    end
+
+    it "returns posts url" do
+      post = FactoryGirl.create(:internal_post)
```
Comment (FerPerales) [inline spec/helpers/post_location_helper_spec.rb]: You can remove the instantiation of the post into a _let_:

``` ruby
describe "#link_to_post" do
  let(:external_post) { FactoryGirl.create(:external_post) }
  let(:internal_post) { FactoryGirl.create(:internal_post) }

  it "Returns gamedev url" do
    ...
...
end

```

File: config/environments/development.rb [hunk 4]
```diff
@@ -14,8 +14,19 @@
...
-  config.action_mailer.default_url_options = { host: 'rubygamedev.dev' }
+  config.action_mailer.raise_delivery_errors = true
+  #config.action_mailer.default_url_options = { host: "localhost:3000" }
+  config.action_mailer.delivery_method = :smtp
+  config.action_mailer.smtp_settings = {
+    address: "smtp.gmail.com",
+    port: 25,
+    domain: "gmail.com",
+    user_name: "powercodegirls@gmail.com",
+    password: "powercode123",dig
```
Comment (sandrely2611-zz) [inline config/environments/development.rb]: this config it's for my lap, sorry the user_name and pass aren't necessary

**Comment by KarenVentura** [comment]
@anymoto  @FerPerales  all is already :+1:

File: app/mailers/application_mailer.rb [hunk 5]
```diff
@@ -0,0 +1,3 @@
+class ApplicationMailer < ActionMailer::Base
+  default from: "from@example.com"
```
Comment (FerPerales) [inline app/mailers/application_mailer.rb]: Let's make this configurable from an ENV variable :)

File: app/models/comment.rb [hunk 6]
```diff
@@ -4,4 +4,9 @@ class Comment < ActiveRecord::Base
 
   validates_presence_of :user
   validates_presence_of :parent
+
+  after_create :notification_mailer
```
Comment (FerPerales) [inline app/models/comment.rb]: The name of the method can be better and more descriptive :wink:

File: app/models/tag.rb [hunk 7]
```diff
@@ -1,6 +1,6 @@
 class Tag < ActiveRecord::Base
   belongs_to :user
   has_and_belongs_to_many :posts
-
   validates_presence_of :user, :title
+  validates :title, uniqueness: true
```
Comment (anymoto) [inline app/models/tag.rb:5]: We added this validation just for testing, I'm not sure it should be there. What do you think @FerPerales ?

Comment (FerPerales) [inline app/models/tag.rb:5]: I think it makes sense to not create duplicated tags, but we may check this with Andrew to make sure this is the expected behaviour

**Comment by FerPerales** [comment]
LGTM

**Pull request merged**